diff --git a/ports/rp2/main.c b/ports/rp2/main.c
index 3b4d351a7..f8ac068c4 100644
--- a/ports/rp2/main.c
+++ b/ports/rp2/main.c
@@ -53,6 +53,7 @@
 #include "pico/unique_id.h"
 #include "hardware/rtc.h"
 #include "hardware/structs/rosc.h"
+#include "hardware/vreg.h"
 #if MICROPY_PY_LWIP
 #include "lwip/init.h"
 #include "lwip/apps/mdns.h"
@@ -74,6 +75,12 @@ bi_decl(bi_program_feature_group_with_flags(BINARY_INFO_TAG_MICROPYTHON,
     BI_NAMED_GROUP_SEPARATE_COMMAS | BI_NAMED_GROUP_SORT_ALPHA));
 
 int main(int argc, char **argv) {
+    // Apply a modest overvolt, default is 1.10v.
+    // this is required for a stable 250MHz on some RP2040s
+    vreg_set_voltage(VREG_VOLTAGE_1_20);
+    sleep_ms(10);
+    set_sys_clock_khz(250000, true);
+
     // This is a tickless port, interrupts should always trigger SEV.
     SCB->SCR |= SCB_SCR_SEVONPEND_Msk;
 
